packages/offix-datastore/src/models.ts

```diff
@@ -1,21 +1,41 @@
-import { GraphQLSchema, GraphQLObjectType } from "graphql";
-import { parseMarker } from "graphql-metadata";
-import { getUserTypesFromSchema } from "@graphql-toolkit/common";
+import { Storage, StoreChangeEvent } from "./storage";
+import { createPredicate } from "./predicates";
 
-export interface Model {
-    __typename: string;
-}
+export type EventTypes = "ADD" | "UPDATE" | "DELETE";
 
-export interface PersistedModel extends Model {
-    id: string;
-}
+export class Model<T> {
+    constructor(
+        private storeName: string,
+        private schema: any,
+        private getStorage: () => Storage
+    ) {}
+
+    save(input: T): Promise<T> {
+        return this.getStorage().save(this.storeName, input);
+    }
+
+    query(predicateFunction?: Function) {
```

Comment:
No strong typing on predicates. I recomend to create issue to discuss how query/filtering will work comparing to server side capabilities. 
You have mentioned some limitations here.

Reply:
Query and filtering will be on separate issue and PR